### DISCUSSION TITLE
New GLib warnings when entering Trash

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2521] [General]
Here I see these warnings when entering Trash from pcmanfm-qt or LXQt file dialog (Qt5 or Qt6 makes no difference), provided that I open the app inside a terminal (e.g., QTerminal):

```
(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: GFileInfo created without standard::size

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: file ../glib/gio/gfileinfo.c: line 1846 (g_file_info_get_size): should not be reached

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: GFileInfo created without standard::is-symlink

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: file ../glib/gio/gfileinfo.c: line 1677 (g_file_info_get_is_symlink): should not be reached

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: GFileInfo created without standard::is-hidden

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: file ../glib/gio/gfileinfo.c: line 1633 (g_file_info_get_is_hidden): should not be reached

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: GFileInfo created without standard::is-backup

(process:951520): GLib-GIO-CRITICAL **: 20:14:44.159: file ../glib/gio/gfileinfo.c: line 1655 (g_file_info_get_is_backup): should not be reached
```

I don't know when they started to show up — GLib's version is 2.78.4 here — but they don't seem to have anything to do with our codes.

EDIT: The messages are probably related to GVFS, because they show up only on entering directories like `trash:///` and `computer:///`.

**Comment by stefonarch** [comment]
Same here:`  journalctl -b | grep -i gfileinfo`

Related:
https://github.com/mate-desktop/caja/issues/1704
https://gitlab.gnome.org/GNOME/nautilus/-/merge_requests/1143

**Reply by tsujan** [reply]
@stefonarch 
Would you please check that https://github.com/lxqt/libfm-qt/pull/778/commits/6f8816b0794d7d6c94d2b4c4e32cf17097a63e7e silences all warning? Just compile the latest Qt6 branch.

EDIT: Only the Qt6 branch of libfm-qt needs to be compiled.

**Reply by stefonarch** [reply]
It does indeed, no messages anymore.

**Reply by tsujan** [reply]
Thanks!

**Comment by tsujan** [comment]
Since we won't release another Qt5-based libfm-qt, non-rolling distros that will remain with Qt5 in their next version *but see the same warnings* can use https://github.com/lxqt/libfm-qt/commit/6f8816b0794d7d6c94d2b4c4e32cf17097a63e7e — it can be applied to the Qt5-based libfm-qt too.

**Comment by tsujan** [comment]
Whatever reason might be behind recent changes to GLib (`gfileinfo.c`), they were made very inconsistently. You've had specific functions for a long time, but now you need to use more generic functions to avoid warnings.

Also, see https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3261 (particularly, Ondrej Holy's comment on that page and the changes he made to Nautilus at https://gitlab.gnome.org/GNOME/nautilus/-/commit/b2057188708561cfdedce4e1361d7d0372d8f04b, which I followed to remove the warnings from libfm-qt).
